base/server/src/com/netscape/cms/servlet/csadmin/Configurator.java

```diff
@@ -409,15 +409,39 @@ public KeyPair createRSAKeyPair(String tag, CryptoToken token, String keySize)
 
         logger.debug("Configurator.createRSAKeyPair(" + token + ")");
 
+        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages[] = {
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.ENCRYPT,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.DECRYPT,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.WRAP,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.UNWRAP,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN_RECOVER
+        };
+
+        org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage usages_mask[] = {
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.ENCRYPT,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.DECRYPT,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.WRAP,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.UNWRAP,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN,
+            org.mozilla.jss.crypto.KeyPairGeneratorSpi.Usage.SIGN_RECOVER
+        };
```

Comment:
Are these meant to be constants? If so I'd suggest converting them into class/instance attributes.